dcpy/extract/metadata.py

```diff
@@ -28,12 +30,34 @@ class Template(
     library_dataset: LibraryConfig | None = None
 
 
-def read_template(dataset: str, **kwargs) -> Template:
-    file = TEMPLATE_DIR / f"{dataset}.yml"
+def get_jinja_vars(s: str) -> set[str]:
+    env = jinja2.Environment()
+    parsed_content = env.parse(s)
+    return meta.find_undeclared_variables(parsed_content)
+
+
+def read_template(
+    dataset: str, version: str | None = None, template_dir: Path = TEMPLATE_DIR
+) -> Template:
+    file = template_dir / f"{dataset}.yml"
     logger.info(f"Reading template from {file}")
     with open(file, "r") as f:
         template_string = f.read()
-    template_string = jinja2.Template(template_string).render(**kwargs)
+    vars = get_jinja_vars(template_string)
+    if version is None and len(vars) > 0:
+        if vars == {"version"}:
+            raise Exception(
+                "Version must be supplied explicitly to be rendered in template"
+            )
+        else:
+            raise Exception(f"Unsupported jinja vars found in template: {vars}")
+    else:
+        if not (len(vars) == 0 or vars == {"version"}):
```

Comment:
I personally find not-or conditions hard to digest at first. Vs. `if len(vars) > 0 and vars != {"version"}`, but that maybe just be a me thing

Reply:
I'm with you

Reply:
Addressed [here](https://github.com/NYCPlanning/data-engineering/pull/654/commits/989b49f1f8b43a31116b85bceda5e5457240545d)